src/engine/renderer/tr_shader.cpp

```diff
@@ -5303,20 +5484,16 @@ static void SortNewShader()
 	tr.sortedShaders[ i + 1 ] = newShader;
 }
 
-/*
-====================
-GeneratePermanentShader
-====================
-*/
-static shader_t *GeneratePermanentShader()
+// Copy the current global shader to a newly allocated shader.
+static shader_t *MakeShaderPermanent()
```

Comment:
Maybe SetStagesRenderers should be called from here since they are always used together? You could name this function FinalizeShader

Reply:
The function calling this code is already named `FinishShader`, this would be confusing if `FinishShader` calls `FinalizeShader`.

Those functions are basically cutting the large `FinishShader` function in smaller blocks.